python/pyspark/sql/dataframe.py

```diff
@@ -2646,67 +2647,147 @@ def join(
 
         Examples
         --------
-        The following performs a full outer join between ``df1`` and ``df2``.
+        The following examples demonstrate various join types between ``df1`` and ``df2``.
```

Comment:
nit: it's not just df1 and df2 (no need to fix unless other changes)

Reply:
I am neutral on this change here.

Overall, LGTM.

Reply:
Good catch!